pkg/kv/kvserver/store.go

```diff
@@ -1957,6 +1985,104 @@ func (s *Store) systemGossipUpdate(sysCfg *config.SystemConfig) {
 	})
 }
 
+// onSpanConfigUpdate is the callback invoked whenever this store learns of a
+// span config update.
+func (s *Store) onSpanConfigUpdate(ctx context.Context, updated roachpb.Span) {
+	if !spanconfigstore.EnabledSetting.Get(&s.ClusterSettings().SV) {
+		return
+	}
+
+	now := s.cfg.Clock.NowAsClockTimestamp()
+	if err := s.mu.replicasByKey.VisitKeyRange(
+		ctx,
+		keys.MustAddr(updated.Key),
+		keys.MustAddr(updated.EndKey),
```

Comment:
While I don't foresee any reason why we'd get a span config entry for an unaddressable key, I don't see any good reason to panic here as opposed to taking a more defensive step like logging an error and ignoring the update. 

Consider:

```go
sp, err := keys.SpanAddr(updated)
if err != nil {
   log.Errorf(ctx, "...")
   return 
}
if err := s.mu.replicasByKey.VisitKeyRange(ctx, sp.Key, sp.EndKey, func(
	ctx context.Context, it replicaOrPlaceholder,
) error {
    // ...
}; err != nil {
    // ...
}
```